geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/package.scala

```diff
@@ -13,18 +13,38 @@ import org.locationtech.geomesa.arrow.vector.SimpleFeatureVector.SimpleFeatureEn
 import org.locationtech.geomesa.features.serialization.ObjectType.ObjectType
 import org.locationtech.geomesa.utils.conf.GeoMesaSystemProperties.SystemProperty
 import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.io.CloseWithLogging
 import org.opengis.feature.simple.SimpleFeatureType
 
 package object arrow {
 
+  @deprecated("Use Allocator.apply() and clean up afterwards")
   implicit val allocator: BufferAllocator = new RootAllocator(Long.MaxValue)
 
+  // noinspection ScalaDeprecation
+  sys.addShutdownHook(CloseWithLogging(allocator))
+
   // need to be lazy to avoid class loading issues before init is called
   lazy val ArrowEncodedSft: SimpleFeatureType =
     SimpleFeatureTypes.createType("arrow", "batch:Bytes,*geom:Point:srid=4326")
 
+  object ArrowAllocator {
+
+    private val root = new RootAllocator(Long.MaxValue)
```

Comment:
Does this provide a reference to the implicit BufferAllocator above or is it a different one?

Reply:
this is a different one

Reply:
I changed the implicit one to be lazy so it won't normally be instantiated unless it's being used by 3rd party code